src/main/core/Subscriber/FunctionalEventSubscriber.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Claroline\CoreBundle\Subscriber;
+
+use Claroline\CoreBundle\Entity\Log\FunctionalLog;
+use Claroline\CoreBundle\Event\CatalogEvents\FunctionalEvents;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class FunctionalEventSubscriber implements EventSubscriberInterface
+{
+    private $translator;
+    private $em;
+
+    public function __construct(EntityManagerInterface $em, TranslatorInterface $translator)
+    {
+        $this->em = $em;
+        $this->translator = $translator;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            FunctionalEvents::ADD_BADGE => 'logEvent',
+            FunctionalEvents::REMOVE_BADGE => 'logEvent',
```

Comment:
It's from a plugin, this cannot be managed by the core.

Reply:
Done, thx :)